### PR TITLE
doc: Added documentation for enable_trtllm_sampler.

### PR DESCRIPTION
**Pull request by dcampora**
# PR title

Please write the PR title by following template:

[JIRA ticket link/nvbug link/github issue link][fix/feat/doc/infra/...] \<summary of this PR\>

For example, assume I have a PR hope to support a new feature about cache manager of Jira TRTLLM-1000 ticket, it would be like

[TRTLLM-1000][feat] Support a new feature about cache manager

## Description

Please explain the issue and the solution in short.

## GitHub Bot Help

`/bot [-h] ['run', 'kill', 'skip', 'reuse-pipeline'] ...`

Provide a user friendly way for developers to interact with a Jenkins server.

Run `/bot [-h|--help]` to print this help message.

See details below for each supported subcommand.

<details>

`run  [--disable-fail-fast --skip-test --stage-list "A10-1, xxx" --gpu-type "A30, H100_PCIe" --add-multi-gpu-test --only-multi-gpu-test --disable-multi-gpu-test --post-merge --extra-stage "H100_PCIe-[Post-Merge]-1, xxx"]`

Launch build/test pipelines. All previously running jobs will be killed.

`--disable-fail-fast ` *(OPTIONAL)* : Disable fail fast on build/tests/infra failures.

`--skip-test ` *(OPTIONAL)* : Skip all test stages, but still run build stages, package stages and sanity check stages. Note: Does **NOT** update GitHub check status.

`--stage-list "A10-1, xxx"` *(OPTIONAL)* : Only run the specified test stages. Examples: "A10-1, xxx". Note: Does **NOT** update GitHub check status.

`--gpu-type "A30, H100_PCIe"` *(OPTIONAL)* : Only run the test stages on the specified GPU types. Examples: "A30, H100_PCIe". Note: Does **NOT** update GitHub check status.

`--only-multi-gpu-test ` *(OPTIONAL)* : Only run the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--disable-multi-gpu-test ` *(OPTIONAL)* : Disable the multi-GPU tests. Note: Does **NOT** update GitHub check status.

`--add-multi-gpu-test ` *(OPTIONAL)* : Force run the multi-GPU tests. Will also run L0 pre-merge pipeline.

`--post-merge ` *(OPTIONAL)* : Run the L0 post-merge pipeline instead of the ordinary L0 pre-merge pipeline.

`--extra-stage "H100_PCIe-[Post-Merge]-1, xxx"` *(OPTIONAL)* : Run the ordinary L0 pre-merge pipeline and specified test stages. Examples: --extra-stage "H100_PCIe-[Post-Merge]-1, xxx".

### kill

`kill  `

Kill all running builds associated with pull request.

### skip

`skip --comment COMMENT `

Skip testing for latest commit on pull request. `--comment "Reason for skipping build/test"` is required. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

### reuse-pipeline

`reuse-pipeline `

Reuse a previous pipeline to validate current commit. This action will also kill all currently running builds associated with the pull request. IMPORTANT NOTE: This is dangerous since lack of user care and validation can cause top of tree to break.

</details>

### CODE REVIEW

File: docs/source/torch.md [hunk 1]
```diff
@@ -37,6 +37,27 @@ cd TensorRT-Model-Optimizer/examples/llm_ptq
 scripts/huggingface_example.sh --model <huggingface_model_card> --quant fp8 --export_fmt hf
 ```
 
+## Sampling
+
+The PyTorch backend supports most of the sampling features that are supported on the C++ backend, such as temperature, top-k and top-p sampling, stop words, bad words, penalty, context and generation logits, and log probs.
```
Comment (a-mccarthy) [inline docs/source/torch.md:42]: is this https://nvidia.github.io/TensorRT-LLM/advanced/speculative-decoding.html related to this content? Wondering if there is a place that you can link to for folks to learn more about sampling. Or what each of these sampling features are?

File: docs/source/torch.md [hunk 2]
```diff
@@ -37,6 +37,27 @@ cd TensorRT-Model-Optimizer/examples/llm_ptq
...
+sampling_params = SamplingParams(
+        temperature=1.0,
+        top_k=8,
+        top_p=0.5,
+    )
+llm.generate(["Hello, my name is",
+            "Hello, my name is"], sampling_params)
+```
+
+When using speculative decoders such as MTP or Eagle-3, the option `enable_trtllm_sampler` is not yet available and therefore the subset of sampling options available is more restricted. We are planning to extend support for speculative decoders in the future.
```
Comment (a-mccarthy) [inline docs/source/torch.md:59]:
```suggestion
When using speculative decoders such as MTP or Eagle-3, the `enable_trtllm_sampler` option is not yet supported and therefore the subset of sampling options available is more restricted. 
```

recommendation to avoid mention future features/support in docs, just in case timelines or priorities change.

Its also not very clear to me what sampling options are still available?

**Review by a-mccarthy — COMMENTED** [review]
just a few questions to possibly help with clarity

File: docs/source/torch.md (hunk 1 above)
Comment (dcampora) [reply docs/source/torch.md:42]: The link you suggest is the set of features supported by the C++ backend speculative decoding. These are not supported at the moment in the PyTorch backend.

I could not find a place where these sampling features are explained. We could add such a page, I would need to find the time to write it.

**Comment by dcampora** [comment]
@a-mccarthy this should be ready to merge, in case you have no further comments, could you please approve it? Thanks a lot

**Review by a-mccarthy — APPROVED** [review]
thanks @dcampora!

**Review by QiJune — APPROVED** [review]
